pandas/core/tools/timedeltas.py

```diff
@@ -118,8 +122,18 @@ def to_timedelta(arg, unit=None, errors="raise"):
             "arg must be a string, timedelta, list, tuple, 1-d array, or Series"
         )
 
-    if isinstance(arg, str) and unit is not None:
-        raise ValueError("unit must not be specified if the input is/contains a str")
+    if isinstance(arg, str):
+        if unit is not None:
+            raise ValueError(
+                "unit must not be specified if the input is/contains a str"
+            )
+        elif arg.upper().endswith(" M") or arg.upper().endswith(" Y"):
```

Comment:
As mentioned in your other commented, we also want to catch arguments without the space as well e.g. `"1Y"`. 

Also we need a test for this warning

Reply:
Done